frontend/packages/pipelines-plugin/src/components/repository/RepositoryDetails.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { GithubIcon } from '@patternfly/react-icons';
+import { useTranslation } from 'react-i18next';
+import { SectionHeading, ResourceSummary, ExternalLink } from '@console/internal/components/utils';
+import { RepositoryKind } from './types';
+
+export interface RepositoryDetailsProps {
+  obj: RepositoryKind;
+}
+
+const RepositoryDetails: React.FC<RepositoryDetailsProps> = ({ obj: repository }) => {
+  const { t } = useTranslation();
+  const { spec } = repository;
+
+  return (
+    <div className="co-m-pane__body">
+      <SectionHeading text={t('pipelines-plugin~Repository details')} />
+      <div className="row">
+        <div className="col-sm-6">
+          <ResourceSummary resource={repository} />
+        </div>
+        <div className="col-sm-6">
+          <dl>
+            <dt>{t('pipelines-plugin~Repository')}</dt>
```

Comment:
I know there isn't real validation on the backend, so for now I suggest add the checks here like other places.
If spec url is not present, this is how it looks now
![image](https://user-images.githubusercontent.com/9964343/129198027-6a5365e1-19e4-418a-90c5-cb475cb9ecf0.png)

Reply:
If the `spec` itself is not there, then dont show the right side column altogether and you could add this to your test case

Reply:
added spec?.repo check

Reply:
updated